service/controller/clusterapi/v29/resource/clusterazs/create.go

```diff
@@ -15,6 +20,18 @@ import (
 	"github.com/giantswarm/aws-operator/service/controller/clusterapi/v29/key"
 )
 
+// subnetPair is temporary type for mapping existing subnets from
+// controllercontext to AZs.
+type subnetPair struct {
+	public  net.IPNet
+	private net.IPNet
+}
+
+func (sp subnetPair) areEmpty() bool {
+	return (sp.public.IP == nil && sp.public.Mask == nil) && (sp.private.IP == nil && sp.private.Mask == nil)
+
+}
+
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```

Comment:
The functionality here looks so complicated we should have unit tests for this to also see some input/output. I have really issues comprehending the code and what it does.

Reply:
Makes sense.

Reply:
I split up the logic into smaller functions and wrote bunch of tests for them.